src/amqpvalue.c

```diff
@@ -4963,7 +4963,6 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                             if (list_item == NULL)
                             {
                                 internal_decoder_data->decoder_state = DECODER_STATE_ERROR;
-                                result = __FAILURE__;
```

Comment:
>result = __FAILURE__; [](start = 32, length = 21)

Looking at this I actually think we should return failure in these cases. One easy way would be in ln 4980 an extra check for internal_decoder_data->decoder_state == DECODER_STATE_ERROR.

Something like:
if (
  (internal_decoder_data->inner_decoder == NULL) ||
  (internal_decoder_data->decoder_state == DECODER_STATE_ERROR)
)

Same applies for the places below.